tests/containers/testenv_validation.pm

```diff
@@ -0,0 +1,39 @@
+# SUSE's openQA tests
+#
+# Copyright 2009-2013 Bernhard M. Wiedemann
+# Copyright 2013-2024 SUSE LLC
+# SPDX-License-Identifier: FSFAP
+
+# Package: podman engine
+# Summary: Verify Podman nginx container is still running after upgrade
+# Maintainer: QE-C team <qa-c@suse.de>
+
+use Mojo::Base 'containers::basetest';
+use testapi;
+use serial_terminal 'select_serial_terminal';
+use utils;
+use containers::common;
+use containers::utils;
+use containers::container_images;
+
+
+sub run {
+    my ($self) = @_;
+    my $unit_name = 'my_nginx';
+    my $container_name = 'nginx-Quadlet';
+    select_serial_terminal;
+
+    my $podman = $self->containers_factory('podman');
+    $self->{podman} = $podman;
+
+
+    # validate the container
+    my $target_version = get_var("TARGET_VERSION", get_required_var("VERSION"));
+    assert_script_run("grep VERSION= /etc/os-release | grep $target_version");
```

Comment:
I would remove the grep here and use the `validate_script_output` with regex.

Reply:
Agree. `validate_script_output` has the advantage, that you see the output in openQA. This means that in the error case you still see what the output was, as opposed to `assert_script_run(' ... | grep ... ')` that swallows the output and makes it opaque.